<launch>
  <!-- TF names-->


  <arg name="dataOutputDir"  default="$(find handeye_calib_sphere)/dataset" />


  <arg name="robotPoseOutputFileName"   default="robot_pose.xml" />

  <!-- End Effector Transform, the ros topic for the tf frame
       at the tip of your robot's gripper or other tool -->
  <arg name="EETF"              default="/arm_left_link_7_t" />

  <!-- Robot Base Transform, the ros topic for the base tf frame of your robot -->
  <arg name="baseTF"            default="/base_link" />

  
  <arg name="cameraIntrinsicInput"  default="$(find handeye_calib_sphere)/camera_intrinsic_color_realsense2.xml" />


  <arg name="RGBDserviceName"  default="kinect_server" />

  <node pkg="handeye_calib_sphere" type="grab_data_node" name="grab_data_node" output="screen">
  <!-- handeye_calib_camodocal arg pass -->
      <param name="robotPoseOutput"      type="str"  value="$(arg dataOutputDir)/$(arg robotPoseOutputFileName)" />
      <param name="cameraIntrinsicInput" type="str"  value="$(arg cameraIntrinsicInput)" />
      <param name="RGBDserviceName"      type="str"  value="$(arg RGBDserviceName)" />
      <param name="dataOutputDir"        type="str"  value="$(arg dataOutputDir)" />
      <param name="EETF"                 type="str"  value="$(arg EETF)" />
      <param name="baseTF"               type="str"  value="$(arg baseTF)" />


      <param name="useQrExtractor"       type="bool" value="true"/>
      <param name="recordingTF"          type="bool" value="true"/>
      <param name="numOfQRcodeOnSide"    type="int"  value="6" />
  </node>

</launch>
